test/y2add_on/clients/add-on_auto_test.rb

```diff
@@ -3,20 +3,237 @@
 require_relative "../../test_helper"
 require "add-on/clients/add-on_auto"
 
+require "byebug"
+
 Yast.import "Packages"
 
 describe Yast::AddOnAutoClient do
-  describe "#main" do
+  describe "#run" do
     before do
       allow(Yast::WFM).to receive(:Args).with(no_args).and_return([func])
       allow(Yast::WFM).to receive(:Args).with(0).and_return(func)
     end
 
+    context "when 'func' is 'Import'" do
```

Comment:
what I usually did is that I believe AutoClient works and simply test directly import method, but it is up to you.